lib/analyse_ocamlformat.mli

```diff
@@ -1,2 +1,8 @@
+type source =
+  | Opam of { version : string } (** Should install OCamlformat from Opam. *)
+  | Vendored of { path : string } (** OCamlformat is vendored. [path] is relative to the project's root. *)
+[@@deriving yojson,eq]
 
-val get_ocamlformat_version : Current.Job.t -> Fpath.t -> string option Lwt.t
+val get_ocamlformat_source : Current.Job.t -> opam_files:string list -> Fpath.t -> source option Lwt.t
+(** Detect the required version of ocamlformat or if it's vendored.
+    Vendored OCamlformat is detected by looking at file names in [opam_files]. *)
```

Comment:
```suggestion
    Vendored OCamlformat is detected by inspecting the [opam_files] paths, which must be specified relative to the given root directory. *)
```
Still needs more explanation to make it clear what `opam_files` really is, and I still think a label on `Fpath.t` is a good idea :slightly_smiling_face:

Reply:
This function doesn't require that the path is relative.
I agree `~root` label feels better.